processor/resourcedetectionprocessor/internal/aws/eks/detector.go

```diff
@@ -62,7 +61,7 @@ var _ detectorUtils = (*eksDetectorUtils)(nil)
 // NewDetector returns a resource detector that will detect AWS EKS resources.
 func NewDetector(set processor.CreateSettings, _ internal.DetectorConfig) (internal.Detector, error) {
 	utils, err := newK8sDetectorUtils()
-	return &detector{utils: utils, logger: set.Logger, err: err}, nil
+	return &detector{utils: utils, logger: set.Logger}, err
```

Comment:
I don't know that this is the correct approach as returning an error here [would prevent](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/resourcedetectionprocessor/internal/resourcedetection.go#L86) all configured detectors from working.  Instead, `Detect()` should probably short-circuit and return this error at that time if there is an error.

Reply:
This error is a failure on start as the configuration is not set properly, and I think the correct approach there is to return right away. Otherwise, `Detect()` will return an error repeatedly every time it executes, flooding the logs.